src/handler/http/request/folders.rs

```diff
@@ -32,25 +55,32 @@ use crate::service::folders;
         ("org_id" = String, Path, description = "Organization name"),
     ),
     request_body(
-        content = Folder,
+        content = CreateFolderRequestBody,
         description = "Folder details",
         example = json!({
             "name": "Infrastructure",
             "description": "Traffic patterns and network performance of the infrastructure",
         }),
     ),
     responses(
-        (status = StatusCode::CREATED, description = "Folder created", body = Folder),
+        (status = StatusCode::OK, description = "Folder created", body = CreateFolderResponseBody),
```

Comment:
As mentioned [here](https://github.com/openobserve/openobserve/pull/5206/files#r1865952760) the old docs for this endpoint were not accurate. When the route succeeded it returned a 200 (OK) response code rather than a 201 (CREATED) response code.

This PR maintains the old behavior of the endpoint but updates the docs to reflect its actual behavior.